stats.py

```diff
@@ -0,0 +1,48 @@
+class WeatherStats:
+    @staticmethod
+    def find_extremes(weather_data):
+        highest_temp = None
+        lowest_temp = None
+        most_humid_day = None
+        highest_temp_day = ""
+        lowest_temp_day = ""
+        most_humid_day_date = ""
+
+        for data in weather_data:
+            if data.max_temp is not None:
+                if highest_temp is None or data.max_temp > highest_temp:
+                    highest_temp = data.max_temp
+                    highest_temp_day = data.date
```

Comment:
Instead of storing these in separate varaibles, use Weatherdata Class instead. To make it more manageable and readable